packages/techdocs-common/src/stages/publish/azureBlobStorage.ts

```diff
@@ -0,0 +1,250 @@
+    const storageClient = new BlobServiceClient(
+      `https://${account}.blob.core.windows.net`,
+      credential,
+    );
+
+    await storageClient
+      .getContainerClient(containerName)
+      .getProperties()
+      .then(() => {
+        logger.info(
+          `Successfully connected to the Azure Blob Storage container ${containerName}.`,
+        );
+      })
+      .catch(reason => {
+        logger.error(
+          `Could not retrieve metadata about the Azure Blob Storage container ${containerName}. ` +
+            'Make sure the Azure project and the container exists and the access key located at the path ' +
+            "techdocs.publisher.azureBlobStorage.credentials defined in app config has the role 'Storage Object Creator'. " +
+            'Refer to https://backstage.io/docs/features/techdocs/using-cloud-storage',
+        );
+        throw new Error(
+          `from Azure Blob Storage client library: ${reason.message}`,
+        );
+      });
+
+    return new AzureBlobStoragePublish(storageClient, containerName, logger);
+  }
+
+  constructor(
+    private readonly storageClient: BlobServiceClient,
+    private readonly containerName: string,
+    private readonly logger: Logger,
+  ) {
+    this.storageClient = storageClient;
+    this.containerName = containerName;
+    this.logger = logger;
+  }
+
+  /**
+   * Upload all the files from the generated `directory` to the Azure Blob Storage container.
+   * Directory structure used in the container is - entityNamespace/entityKind/entityName/index.html
+   */
+  async publish({ entity, directory }: PublishRequest): Promise<void> {
+    try {
+      // Note: Azure Blob Storage manages creation of parent directories if they do not exist.
+      // So collecting path of only the files is good enough.
+      const allFilesToUpload = await getFileTreeRecursively(directory);
+
+      const uploadPromises: Array<Promise<BlobUploadCommonResponse>> = [];
+
+      // Bound the number of concurrent batches. We want a bit of concurrency for
+      // performance reasons, but not so much that we starve the connection pool
+      // or start thrashing.
+      const limiter = limiterFactory(BATCH_CONCURRENCY);
+
+      const promises = allFilesToUpload.map(filePath => {
+        // Remove the absolute path prefix of the source directory
+        // Path of all files to upload, relative to the root of the source directory
+        // e.g. ['index.html', 'sub-page/index.html', 'assets/images/favicon.png']
+        const relativeFilePath = filePath.replace(`${directory}/`, '');
+        const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
+        const destination = path.normalize(
+          `${entityRootDir}/${relativeFilePath}`,
+        ); // Azure Blob Storage Container file relative path
+
+        // TODO: Upload in chunks of ~10 files instead of all files at once.
```

Comment:
Remove this TODO since p-limit has been implemented.

Reply:
Removed